tests/Http/Controllers/ProfileTest.php

```diff
@@ -0,0 +1,434 @@
+
+        $this->actingAs($admin)
+            ->getJson(route('profile.edit', $user->id))
+            ->assertOk();
+    }
+
+    /** @test */
+    public function user_can_see_own_profile_edit_view()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->getJson(route('profile.edit', $user->id))
+            ->assertOk();
+    }
+
+    /** @test */
+    public function user_cannot_see_others_profile_edit_view()
+    {
+        $user1 = User::factory()->create([
+            'is_admin' => false
+        ]);
+        $user2 = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user1)
+            ->getJson(route('profile.edit', $user2->id))
+            ->assertRedirect('/');
+    }
+
+    /** @test */
+    public function guest_cannot_see_profile_edit_view()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->getJson(route('profile.reviews', $user->id))
+            ->assertStatus(401); //unauthorised
+    }
+
+    /** @test */
+    public function admin_can_update_own_profile()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->putJson(route('profile.update', $admin->id), [
+              'name' => 'Admin Generic',
+              'avatar' => 'https://genericAvatars.com',
+              'about_me' => 'This is a sample bit about me'
+            ])->assertStatus(302);
+
+        $this->assertDatabaseHas('users', [
+            'id' => $admin->id,
+            'name' => 'Admin Generic',
+            'email' => $admin->email,
+            'password' => $admin->password,
+            'avatar' => 'https://genericAvatars.com',
+            'about_me' => 'This is a sample bit about me',
+            'is_admin' => $admin->is_admin
+        ]);
+    }
+
+    /** @test */
+    public function admin_cannot_update_others_profile()
+    {
+        $admin = User::factory()->admin()->create();
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($admin)
+            ->putJson(route('profile.update', $user->id), [
+                'name' => 'Joe Generic',
+                'avatar' => 'https://genericAvatars.com',
+                'about_me' => 'This is a sample bit about me'
+            ])->assertStatus(302);
+
+        $this->assertDatabaseMissing('users', [
+            'id' => $user->id,
+            'name' => 'Joe Generic',
+            'email' => $user->email,
+            'password' => $user->password,
+            'avatar' => 'https://genericAvatars.com',
+            'about_me' => 'This is a sample bit about me',
+            'is_admin' => $user->is_admin
+        ]);
+    }
+
+    /** @test */
+    public function user_can_update_own_profile()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
```

Comment:
for update test, the best is to specify the values before the update and then test if they were updated.